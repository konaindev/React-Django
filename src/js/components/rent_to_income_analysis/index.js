import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import { formatCurrencyShorthand, formatPercent } from '../../utils/formatters';

import './rent_to_income_analysis.scss';

const getCategoryIndexOfRate = (categories, rate) =>
  rate ? categories.findIndex((category) => category.low <= rate && category.high > rate) : -1;

export const RentToIncomeAnalysis = ({
  categories,
  incomes,
  rental_rates,
  data
}) => (
  <div className="rent-to-income-analysis">
    <div className="rent-to-income-analysis__title">
      Rent To Income Analysis
    </div>
    <div className="rent-to-income-analysis__panel">
      <div className="rent-to-income-analysis__panel-left">
        <div className="rent-to-income-analysis__ratio">
          Rent to Income Ratio:
        </div>
        <div className="rent-to-income-analysis__categories">
          {categories.map((category, index) => (
            <div key={index} className="rent-to-income-analysis__category">
              <span
                className={cn(
                  'rent-to-income-analysis__category-color',
                  `rent-to-income-analysis__category-color--${index}`
                )}
              />
              <span className="rent-to-income-analysis__category-name">
                {category.name}
              </span>
            </div>
          ))}
        </div>
      </div>
      <div className="rent-to-income-analysis__panel-right">
        <div className="rent-to-income-analysis-chart">
          <div className="rent-to-income-analysis-chart__vertical-caption">
            Monthly Rental (USD)
          </div>
          <div className="rent-to-income-analysis-chart__horizontal-caption">
            Annual Income (USD)
          </div>
          <div className="rent-to-income-analysis-chart__body">
            <div className="rent-to-income-analysis-chart__group rent-to-income-analysis-chart__group--yaxis">
              <div className="rent-to-income-analysis-chart__cell">{' '}</div>
              {rental_rates.map((rentalRate, rateIndex) => (
                <div
                  key={rateIndex}
                  className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis">
                  {formatCurrencyShorthand(rentalRate)}
                </div>
              ))}
            </div>
            {data.map((group, groupIndex) => (
              <div key={groupIndex} className="rent-to-income-analysis-chart__group">
                <div className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--xaxis">
                  {formatCurrencyShorthand(incomes[groupIndex])}
                </div>
                {group.map((rate, rateIndex) => (
                  <div
                    key={`${groupIndex}-${rateIndex}`}
                    className={cn(
                      'rent-to-income-analysis-chart__cell',
                      `rent-to-income-analysis__category-color--${getCategoryIndexOfRate(categories, rate)}`
                    )}>
                    {rate && formatPercent(rate)}
                  </div>
                ))}
              </div>
            ))}
          </div>
        </div>
      </div>
    </div>
  </div>
);

RentToIncomeAnalysis.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    low: PropTypes.number,
    high: PropTypes.number
  })),
  incomes: PropTypes.arrayOf(PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string
  ])).isRequired,
  rental_rates: PropTypes.arrayOf(PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string
  ])).isRequired,
  data: PropTypes.arrayOf(PropTypes.array).isRequired
};

export default RentToIncomeAnalysis;
