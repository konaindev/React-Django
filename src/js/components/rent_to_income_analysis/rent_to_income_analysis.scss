@import "css/base.scss";

$category-color-0: #273042;
$category-color-1: #867ffe;
$category-color-2: #675efc;
$category-color-3: #5147ff;
$category-color-4: #4035f4;

.rent-to-income-analysis {
    $this: &;

    padding: 20px 0;
    margin-bottom: 20px;

    &__title {
        color: $heading-font-color;
        font-size: 14px;
        line-height: 17px;
        letter-spacing: 1px;
        font-weight: 500;
        padding-bottom: 29px;
        border-bottom: 1px solid #586073;
        margin-bottom: 35px;
        text-transform: uppercase;
    }

    &__panel {
        padding: 70px;
        border: $panel-border;
        border-radius: $panel-border-radius;
        background-color: $panel-background-color;
        box-shadow: $panel-box-shadow;
        display: flex;
        flex-direction: row;

        &-left {
            flex-grow: 0;
            max-width: 33.3333%;
            flex-basis: 33.3333%;
        }

        &-right {
            flex-grow: 0;
            flex-basis: 66.6666%;
        }
    }

    &__ratio {
        font-size: 14px;
        letter-spacing: 0.4px;
        line-height: 16px;
        text-transform: uppercase;
        margin-bottom: 32px;
        white-space: nowrap;
    }

    &__categories {
        display: flex;
        flex-direction: column-reverse;
    }

    &__category {
        margin: 10px 0;
        font-size: 14px;
        line-height: 16px;
        color: #6d778d;
        white-space: nowrap;

        &-color {
            width: 32px;
            height: 20px;
            margin-right: 16px;
            display: inline-block;
            vertical-align: middle;

            &--0 {
                color: #6d778d;
                background-color: $category-color-0;
            }

            &--1 {
                background-color: $category-color-1;
                color: #ffffff;
            }

            &--2 {
                background-color: $category-color-2;
                color: #ffffff;
            }

            &--3 {
                background-color: $category-color-3;
                color: #ffffff;
            }

            &--4 {
                background-color: $category-color-4;
                color: #ffffff;
            }
        }

        &-name {
            display: inline-block;
            vertical-align: middle;
        }
    }

    &-chart {
        position: relative;
        padding-left: 15px;
        padding-bottom: 15px;

        &__vertical-caption {
            font-size: 12px;
            letter-spacing: 1px;
            line-height: 15px;
            text-align: right;
            text-transform: uppercase;
            transform: rotate(-90deg);
            transform-origin: right top;
            position: absolute;
            right: 100%;
            top: 0;
            white-space: nowrap;
            color: #6d778d;
        }

        &__horizontal-caption {
            font-size: 12px;
            letter-spacing: 1px;
            line-height: 15px;
            text-align: right;
            text-transform: uppercase;
            position: absolute;
            right: 0;
            bottom: 0;
            white-space: nowrap;
            color: #6d778d;
        }

        &__body {
            display: flex;
            width: 100%;
            flex-direction: row;
            text-transform: uppercase;
        }

        &__group {
            display: flex;
            flex-direction: column-reverse;
            flex: 1;

            &--yaxis {
                max-width: 65px;
                width: 65px;
            }
        }

        &__cell {
            height: 44px;
            line-height: 44px;
            text-align: center;
            font-size: 14px;

            &--yaxis {
                border-right: 1px solid rgba(61, 69, 85, 0.3);
                text-align: right;
                padding-right: 16px;
            }

            &--xaxis,
            &--yaxis {
                color: #ffffff;
                font-size: 12px;
            }
        }
    }

    &--flip {
        #{$this}__panel {
            display: block;

            &-left {
                display: flex;
                max-width: 100%;
                margin-bottom: 32px;
            }
        }

        #{$this}__categories {
            flex-direction: row-reverse;
        }

        #{$this}__category {
            margin: 0 32px 0 0;
        }

        #{$this}__ratio {
            margin: 0 32px 0 0;
        }

        #{$this}-chart {
            &__vertical-caption {
                right: initial;
                top: initial;
                left: 0;
                bottom: 44px;
                transform-origin: left top;
            }

            &__horizontal-caption {
                right: initial;
                left: 80px;
            }
        }
    }
}
