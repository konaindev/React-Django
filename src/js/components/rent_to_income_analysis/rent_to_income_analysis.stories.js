import React from "react";

import { storiesOf } from "@storybook/react";

import RentToIncomeAnalysis from "./index";
import { props_large, props_small } from "./props";

storiesOf("RentToIncomeAnalysis", module)
  .add("small", () => (
    <div style={{ width: 1320, margin: "80px auto" }}>
      <RentToIncomeAnalysis {...props_small} />
    </div>
  ))
  .add("large", () => (
    <div style={{ width: 1320, margin: "80px auto" }}>
      <RentToIncomeAnalysis {...props_large} />
    </div>
  ));

export default RentToIncomeAnalysis;
