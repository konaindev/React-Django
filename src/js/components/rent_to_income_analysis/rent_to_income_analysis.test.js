import renderer from "react-test-renderer";
import RentToIncomeAnalysis from "./index";
import { props_small, props_large } from "./rent_to_income_analysis.stories";

describe("RentToIncomeAnalysis", () => {
  it("renders in small mode correctly", () => {
    const tree = renderer.create(<RentToIncomeAnalysis {...props_small} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders in large mode correctly", () => {
    const tree = renderer.create(<RentToIncomeAnalysis {...props_large} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
