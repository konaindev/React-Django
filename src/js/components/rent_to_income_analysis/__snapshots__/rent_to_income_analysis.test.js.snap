// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RentToIncomeAnalysis renders in large mode correctly 1`] = `
<div
  className="panel panel--default rent-to-income-analysis rent-to-income-analysis--flip"
>
  <div
    className="rent-to-income-analysis__legends-wrapper"
  >
    <div
      className="rent-to-income-analysis__ratio"
    >
      Rent to Income Ratio:
    </div>
    <div
      className="rent-to-income-analysis__categories"
    >
      <div
        className="rent-to-income-analysis__category"
      >
        <span
          className="rent-to-income-analysis__category-color rent-to-income-analysis__category-color--0"
        />
        <span
          className="rent-to-income-analysis__category-name"
        >
          Low
        </span>
      </div>
      <div
        className="rent-to-income-analysis__category"
      >
        <span
          className="rent-to-income-analysis__category-color rent-to-income-analysis__category-color--1"
        />
        <span
          className="rent-to-income-analysis__category-name"
        >
          Moderately Low
        </span>
      </div>
      <div
        className="rent-to-income-analysis__category"
      >
        <span
          className="rent-to-income-analysis__category-color rent-to-income-analysis__category-color--2"
        />
        <span
          className="rent-to-income-analysis__category-name"
        >
          Target
        </span>
      </div>
      <div
        className="rent-to-income-analysis__category"
      >
        <span
          className="rent-to-income-analysis__category-color rent-to-income-analysis__category-color--3"
        />
        <span
          className="rent-to-income-analysis__category-name"
        >
          Moderately High
        </span>
      </div>
      <div
        className="rent-to-income-analysis__category"
      >
        <span
          className="rent-to-income-analysis__category-color rent-to-income-analysis__category-color--4"
        />
        <span
          className="rent-to-income-analysis__category-name"
        >
          High
        </span>
      </div>
    </div>
  </div>
  <div
    className="rent-to-income-analysis__chart-wrapper"
  >
    <div
      className="rent-to-income-analysis-chart"
    >
      <div
        className="rent-to-income-analysis-chart__vertical-caption"
      >
        Monthly Rental (USD)
      </div>
      <div
        className="rent-to-income-analysis-chart__horizontal-caption"
      >
        Annual Income (USD)
      </div>
      <div
        className="rent-to-income-analysis-chart__body"
      >
        <div
          className="rent-to-income-analysis-chart__group rent-to-income-analysis-chart__group--yaxis"
        >
          <div
            className="rent-to-income-analysis-chart__cell"
          >
             
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $1,000
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $2,000
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $3,000
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $4,000
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $5,000
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $6,000
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $7,000
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $8,000
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $9,000
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $10,000
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $11,000
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $12,000
          </div>
        </div>
        <div
          className="rent-to-income-analysis-chart__group"
        >
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--xaxis"
          >
            $50k
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            24%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--4"
          >
            48%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
        </div>
        <div
          className="rent-to-income-analysis-chart__group"
        >
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--xaxis"
          >
            $100k
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            12%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            24%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--3"
          >
            36%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--4"
          >
            48%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
        </div>
        <div
          className="rent-to-income-analysis-chart__group"
        >
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--xaxis"
          >
            $200k
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            6%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            12%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            18%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            24%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            30%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--3"
          >
            36%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--3"
          >
            42%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--4"
          >
            48%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--4"
          >
            54%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
        </div>
        <div
          className="rent-to-income-analysis-chart__group"
        >
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--xaxis"
          >
            $300k
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            4%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            8%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            12%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            16%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            20%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            24%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            28%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            32%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--3"
          >
            36%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--3"
          >
            40%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--3"
          >
            44%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--4"
          >
            48%
          </div>
        </div>
        <div
          className="rent-to-income-analysis-chart__group"
        >
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--xaxis"
          >
            $400k
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            3%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            6%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            9%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            12%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            15%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            18%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            21%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            24%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            27%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            30%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            33%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--3"
          >
            36%
          </div>
        </div>
        <div
          className="rent-to-income-analysis-chart__group"
        >
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--xaxis"
          >
            $500k
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            2%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            5%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            7%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            10%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            12%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            14%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            17%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            19%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            22%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            24%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            26%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            29%
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RentToIncomeAnalysis renders in small mode correctly 1`] = `
<div
  className="panel panel--default rent-to-income-analysis"
>
  <div
    className="rent-to-income-analysis__legends-wrapper"
  >
    <div
      className="rent-to-income-analysis__ratio"
    >
      Rent to Income Ratio:
    </div>
    <div
      className="rent-to-income-analysis__categories"
    >
      <div
        className="rent-to-income-analysis__category"
      >
        <span
          className="rent-to-income-analysis__category-color rent-to-income-analysis__category-color--0"
        />
        <span
          className="rent-to-income-analysis__category-name"
        >
          Low
        </span>
      </div>
      <div
        className="rent-to-income-analysis__category"
      >
        <span
          className="rent-to-income-analysis__category-color rent-to-income-analysis__category-color--1"
        />
        <span
          className="rent-to-income-analysis__category-name"
        >
          Moderately Low
        </span>
      </div>
      <div
        className="rent-to-income-analysis__category"
      >
        <span
          className="rent-to-income-analysis__category-color rent-to-income-analysis__category-color--2"
        />
        <span
          className="rent-to-income-analysis__category-name"
        >
          Target
        </span>
      </div>
      <div
        className="rent-to-income-analysis__category"
      >
        <span
          className="rent-to-income-analysis__category-color rent-to-income-analysis__category-color--3"
        />
        <span
          className="rent-to-income-analysis__category-name"
        >
          Moderately High
        </span>
      </div>
      <div
        className="rent-to-income-analysis__category"
      >
        <span
          className="rent-to-income-analysis__category-color rent-to-income-analysis__category-color--4"
        />
        <span
          className="rent-to-income-analysis__category-name"
        >
          High
        </span>
      </div>
    </div>
  </div>
  <div
    className="rent-to-income-analysis__chart-wrapper"
  >
    <div
      className="rent-to-income-analysis-chart"
    >
      <div
        className="rent-to-income-analysis-chart__vertical-caption"
      >
        Monthly Rental (USD)
      </div>
      <div
        className="rent-to-income-analysis-chart__horizontal-caption"
      >
        Annual Income (USD)
      </div>
      <div
        className="rent-to-income-analysis-chart__body"
      >
        <div
          className="rent-to-income-analysis-chart__group rent-to-income-analysis-chart__group--yaxis"
        >
          <div
            className="rent-to-income-analysis-chart__cell"
          >
             
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $1,000
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $2,000
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $3,000
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $4,000
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $5,000
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--yaxis"
          >
            $6,000
          </div>
        </div>
        <div
          className="rent-to-income-analysis-chart__group"
        >
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--xaxis"
          >
            $50k
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            24%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--4"
          >
            48%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
        </div>
        <div
          className="rent-to-income-analysis-chart__group"
        >
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--xaxis"
          >
            $100k
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            12%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            24%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--3"
          >
            36%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--4"
          >
            48%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color---1"
          />
        </div>
        <div
          className="rent-to-income-analysis-chart__group"
        >
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--xaxis"
          >
            $200k
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            6%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            12%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            18%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            24%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            30%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--3"
          >
            36%
          </div>
        </div>
        <div
          className="rent-to-income-analysis-chart__group"
        >
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--xaxis"
          >
            $300k
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            4%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            8%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            12%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            16%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            20%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--2"
          >
            24%
          </div>
        </div>
        <div
          className="rent-to-income-analysis-chart__group"
        >
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--xaxis"
          >
            $400k
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            3%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            6%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            9%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            12%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            15%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            18%
          </div>
        </div>
        <div
          className="rent-to-income-analysis-chart__group"
        >
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis-chart__cell--xaxis"
          >
            $500k
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            2%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            5%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            7%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--0"
          >
            10%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            12%
          </div>
          <div
            className="rent-to-income-analysis-chart__cell rent-to-income-analysis__category-color--1"
          >
            14%
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
